ci/ansible/playbooks/os/redhat/upgrade_release.yml

```diff
@@ -0,0 +1,378 @@
+      reboot:
+        msg: Reboot initiated by Ansible due to major release upgrade
+        connect_timeout: "{{ reboot.connect_timeout }}"
+        reboot_timeout: "{{ reboot.upgrade_timeout }}"
+
+    ### POST-UPGRADE -- CLEAN-UP AND VERIFICATION ###
+    - name: Verify post-upgrade state
+      block:
+        - name: Gather post-upgrade facts
+          setup:
+            gather_subset: "!all"
+
+        - name: Assert major version is as expected
+          assert:
+            that:
+              - ansible_distribution_version is version('8.4','=')
+              - ansible_kernel is version('4.18.0-305','>=')
+            quiet: true
+
+        - name: Verify subscription status for non_cloud machines
+          when: provider == "non_cloud"
+          block:
+            - name: Register OS version in subscription-maganer
+              shell: "subscription-manager list --installed|awk '/Version:/ { print $2 }'"
+              register: subscription_version
+
+            - name: Register subscription status
+              shell: "subscription-manager list --installed|awk '/Status:/ { print $2 }'"
+              register: subscription_status
+
+            - name: Check that upgraded version remains correctly subscribed
+              assert:
+                that:
+                  - subscription_version.stdout == "8.4"
+                  - subscription_status.stdout == "Subscribed"
+                quiet: true
+
+    - name: Remove packages from the dnf exclude list # Populated by leapp during upgrade
+      ini_file:
+        path: /etc/dnf/dnf.conf
+        section: main
+        option: exclude
+        value: ''
+
+    - name: Remove leapp_resume service unit
+      block:
+        - name: Disable leapp_resume service # This unit may remain even when the leapp packages are removed
+          systemd:
+            name: leapp_resume
+            state: stopped
+            enabled: false
+            daemon-reload: true
+            force: true
+
+        - name: Remove leapp_resume unit file # Package removal does not remove this units automatically
+          file:
+            path: /etc/systemd/system/leapp_resume.service
+            state: absent
+
+    - name: Remove leapp packages
+      package:
+        name: "leapp*" # Leapp packages may have different nomenclature in RHEL8
+        state: absent
+
+    - name: Remove leapp configuration and log files
+      file:
+        path: /etc/leapp
+        state: absent
+
+    - name: Remove remaining RHEL 7 packages
+      block:
+        - name: Register remaining RHEL7 packages
+          shell: "rpm -qa | grep -e '\\.el7' | grep -vE '^(gpg-pubkey|libmodulemd|katello-ca-consumer)'"
+          register: remaining_rhel7_packages
+          ignore_errors: true
+
+        - name: Remove remaining RHEL7 packages
+          when: not remaining_rhel7_packages.failed
+          dnf:
+            name: "{{ remaining_rhel7_packages.stdout_lines }}"
```

Comment:
Can we specify list of packages here directly using patterns with `{{ package_name }}*.el7*`?